java/common/exception/ErrorMessage.java

```diff
@@ -105,6 +105,8 @@ public class ErrorMessage extends com.vaticle.typedb.common.exception.ErrorMessa
             new ErrorMessage(41, "'%s' is not a valid Type label. Type labels must start with a letter, and may contain only letters, numbers, '-' and '_'.");
     public static final ErrorMessage INVALID_ANNOTATION =
             new ErrorMessage(42, "Invalid annotation '%s' on '%s' constraint");
+    public static final ErrorMessage RELATION_NOT_ANONYMOUS =
```

Comment:
This might fit better close to `INVALID_RULE_THEN_ROLES`.
(This will indeed require you to increase every following key by 1. Unchanging ErrorCodes will hopefully be in TypeDB 3)

Can the name be updated to reflect the fact that it's about  `then` clauses?

Reply:
how about RELATION_IN_THEN_NOT_ANONYMOUS?